src/recorder/jamrecorder.cpp

```diff
@@ -134,6 +139,21 @@ CJamSession::CJamSession(QDir recordBaseDir) :
     vecptrJamClients.fill(nullptr);
 }
 
+/**
+ * @brief CJamSession::~CJamSession
+ */
+CJamSession::~CJamSession()
+{
+    for (int i = 0; i < jamClientConnections.count(); i++ )
+    {
+        if ( jamClientConnections[i] )
+        {
+            delete jamClientConnections[i];
```

Comment:
This feels risky - do we know the connections have completed writing?  Will this wait for them to do so?

Reply:
Maybe there is another check to be made? But certainly we don't want to destroy the `CJamSession` instance without freeing the client connections it is pointing to.

Reply:
Actually, thinking now, this looks okay - it's only called when the CJamSession instance is being destroy _anyway_, so everything it owned should already have finished work - otherwise something already went badly wrong.